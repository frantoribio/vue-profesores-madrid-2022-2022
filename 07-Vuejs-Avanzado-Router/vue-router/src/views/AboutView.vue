<template>
  <div class="about">
    <h1>Hola Profesores de Vue.js</h1>
  </div>
</template>

<style scoped></style>
