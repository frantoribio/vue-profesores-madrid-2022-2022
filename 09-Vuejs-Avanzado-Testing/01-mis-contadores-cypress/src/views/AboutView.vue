<template>
  <main>
    <h1>About</h1>
    <p>About page</p>
    <Modal />
  </main>
</template>

<script setup>
  import Modal from '@/components/Modal.vue'
</script>

<style scoped>
  @media (min-width: 1024px) {
    .about {
      min-height: 100vh;
      display: flex;
      align-items: center;
    }
  }
</style>
