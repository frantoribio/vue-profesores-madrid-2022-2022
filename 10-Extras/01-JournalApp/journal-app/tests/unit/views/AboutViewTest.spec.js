import { describe, expect, test } from 'vitest'

import AboutView from '@/views/AboutView.vue'
import { shallowMount } from '@vue/test-utils'

describe('View -> AboutView', () => {
  test('Debe hacer match con el snapshot', () => {
    const wrapper = shallowMount(AboutView)
    expect(wrapper.html()).toMatchSnapshot() // snapshot es una imagen de la vista y podemos usar .element tambien o html()
  })
})
