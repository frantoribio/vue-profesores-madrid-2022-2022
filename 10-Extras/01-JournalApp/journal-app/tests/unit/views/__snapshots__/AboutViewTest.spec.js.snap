// Vitest Snapshot v1

exports[`View -> AboutView > Debe hacer match con el snapshot 1`] = `
"<!-- Si quiero poner una cosa en mi tema, solo debo ponerle
data-theme=\\"dark\\" -->
<div class=\\"container-md py-4mx-auto rounded-lg px-3 shadow-xl\\">
  <article class=\\"prose mx-auto max-w-3xl text-center\\">
    <h1 class=\\"cursor-pointer text-xl font-semibold text-gray-500 hover:text-indigo-500\\"> This is an about page </h1>
    <h1>Temas</h1>
    <div class=\\"card card-side bg-base-100 shadow-xl\\">
      <figure><img src=\\"https://placeimg.com/200/280/arch\\" alt=\\"Movie\\" class=\\"ml-3 rounded-md shadow-md\\"></figure>
      <div class=\\"card-body\\">
        <h2 class=\\"card-title\\">Temas de colores</h2>
        <p>Click the button to watch on Jetflix app.</p>
        <div class=\\"card-actions justify-end\\"><button class=\\"btn btn-primary\\">Watch</button><button class=\\"btn btn-secondary\\">Watch</button></div>
      </div>
    </div>
    <h2>Â¿Seguro?</h2>
    <p> Lorem, ipsum dolor sit amet consectetur adipisicing elit. Repudiandae minus quam harum ab dolorum laudantium perspiciatis dolores corporis commodi, a quidem. Possimus ipsum magnam repellendus ipsam laboriosam architecto mollitia consectetur?. </p><button data-set-theme=\\"\\" class=\\"btn btn-outline btn-primary\\"> Default </button><button data-set-theme=\\"light\\" class=\\"btn btn-primary\\"> Light </button><button data-set-theme=\\"dark\\" class=\\"btn btn-secondary\\"> Dark </button><button data-set-theme=\\"lemonade\\" class=\\"btn btn-warning\\"> Lemonade </button><button data-set-theme=\\"night\\" class=\\"btn btn-accent\\"> Night </button><button data-set-theme=\\"joseluisgs\\" class=\\"btn btn-outline btn-success\\"> JoseLuisGS </button>
  </article>
</div>"
`;
